### DISCUSSION TITLE
App icons and image resource sizes

### DISCUSSION THREAD

**Discussion by nubilion** [zepp-health/zeppos-forum#47] [Developer Feedback]
Hello, I have a couple of questions regarding the app icon and image resource sizes for different watches:

1. Shall we always use 124x124 px image size of app icon, no matter the screen resolution of the actual watch? For image assets, I know that we have to scale them. However, it is not clear whether we also need to scale them for for the app icons?
2. Could someone explain, how did you calculate the icon sizes for the different screen resolutions, as it is shown in zeppos documentation?
https://docs.zepp.com/docs/1.0/designs/visual/icons/
![image](https://user-images.githubusercontent.com/89732401/211885743-1a4d2f77-c397-458d-8a43-168410daf618.png)

**Comment by htoooth** [comment]
Hi, @nubilion:
App icon specification in documentation is the best way to adapt our devices. And you can also use one resolution to adapt to all devices.

**Reply by nubilion** [reply]
Thank you for the answer. This actually answers my second question. In this case, the actual physical size of the icons used for e.g. buttons in the app, will be different, due to different screen resolution. E.g. 64x64 button would take 64/480 = 13.33% of the GTR3PRO  screen width, whereas the same button of 64x64 px would take  64/390 = 16,41% of the GTS3 screen width. Please confirm whether this is desirable behavior?

I still need an answer to my first question. This is regarding the main app icon, the one which is shown in the watch app launcher. Namely, in the documentation it says that it should be 124x124 px size. Nevertheless, it is not clear whether the same icon size should be used for all smartwatches resolutions, or it should scale with the screen size. Could you please tell me which should be the main app icon size for GST4 mini watch?

**Reply by htoooth** [reply]
Hi, @nubilion 

1. Now our device is without scaling feature. The size and dimensions of the button (64x64) on different screens depends on the developer code implementation. If you don't consider this in your code, the button will always be 64x64 in size.
2. App Icon can be automatically scaled in the launcher. However, the image resources inside the app cannot be scaled, so the image needs to be pre-scaled for each device.

> In this case, the actual physical size of the icons used for e.g. buttons in the app, will be different, due to different screen resolution. E.g. 64x64 button would take 64/480 = 13.33% of the GTR3PRO screen width, whereas the same button of 64x64 px would take 64/390 = 16,41% of the GTS3 screen width. Please confirm whether this is desirable behavior?

I suggest you use 124x124  app icon for all devices. 

> I still need an answer to my first question. This is regarding the main app icon, the one which is shown in the watch app launcher. Namely, in the documentation it says that it should be 124x124 px size. Nevertheless, it is not clear whether the same icon size should be used for all smartwatches resolutions, or it should scale with the screen size. Could you please tell me which should be the main app icon size for GST4 mini watch?

I'm sorry that documentation confuses you. We will optimize the content of the app icon.

**Reply by htoooth** [reply]
@nubilion 
This is my error. I have fixed it.

**Reply by nubilion** [reply]
I still have problem with GTS4 mini smartwatch. When I use icon size of 124x124px as you suggested, I get cropped icon size in the app launcher. I got the same complaint from the app reviewers when I submitted the app. I noticed that this only happens on GTS4 mini smartwatch. Could you please check this issue.
![zepp_screenshot_1673545703487](https://user-images.githubusercontent.com/89732401/212142430-f1ce7350-a138-4eea-b935-dab9f32ca046.png)

**Reply by NavigationWear** [reply]
If may enter this conversation, I have exactly the same problem - complaint from the app reviewers after the last app submission (3 days ago). Hovewer I haven't made any changes in icons or scaling. So I suspect, that it is some new issue in Zepp OS which occured in GTS4 mini.
![image1673422737694](https://user-images.githubusercontent.com/122609345/212307467-b19bbcc3-290c-4609-99e6-2eb26a9ad49f.png)

**Reply by nubilion** [reply]
Thank you for confirmation. It seems that the problem is on the ZeppOS side. I will then submit apps without GTS4 mini in the package. I hope ZeppOS team will solve this issue soon.

**Reply by liqin-ZeppHealth** [reply]
Hi @nubilion @NavigationWear , we are checking this app icon issue on GTS4 mini. and if this is the only issue that block your submission, a possible solution is that you can use a 68x68px app icon for GTS4 mini specifically.

**Comment by MyoungXUE** [comment]
@nubilion We have fixed this issue in the latest firmware of the GTR 4 mini, thanks again for your feedback.
